### PR TITLE
Jdh-8 asm support  #19227

### PR DESCRIPTION
**Pull request by slowhand99**
**Checklist**

- [ ] Closing issues: #issue
- [x] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [radare2book](https://github.com/radareorg/radare2book)

**Description**
##asm

Support disassembly for JDH-8 toy architecture

### CODE REVIEW

**Comment by trufae** [comment]
Can you add it to the meson too?

File: libr/asm/arch/jdh8/jdh8dis.c
Comment (trufae) [inline libr/asm/arch/jdh8/jdh8dis.c:18]: add an "else", after the first if, the second will never be reached

File: libr/asm/arch/jdh8/jdh8dis.c
Comment (trufae) [inline libr/asm/arch/jdh8/jdh8dis.c:27]:
```suggestion
	const ut8 high = (*buf & 0xf0) >> 4;
```

File: libr/asm/arch/jdh8/jdh8dis.c
Comment (trufae) [inline libr/asm/arch/jdh8/jdh8dis.c:35]: sdb_fmt is not thread safe and we are trying to get rid of it across all the r2codebase, there are alternatives like snprintf or r_str_newf in case it must be held of the heap. in this case it would work better to have a local buffer of 64 bytes in this function and use snprintf like this:

```
char buf[64];
buf_asm = snprintf (buf, sizeof (buf), "mw %c, %c", ...);
..
```

Comment (trufae) [reply libr/asm/arch/jdh8/jdh8dis.c:35]: you can add a #define to avoid having to pass buf and sizeof(buf) all the time so its easier for the eye 😄

File: libr/asm/arch/jdh8/jdh8dis.c
Comment (trufae) [inline libr/asm/arch/jdh8/jdh8dis.c:26]: why `rlen`? maybe `ilen` works better as a name as it stands for `instruction length`

**Review by trufae — CHANGES_REQUESTED** [review]
as a suggestion, i think it will be better to make it as an RAnalPlugin, because you can now use analysis plugins from the RAsm api, and reuse the same code when you implement the code analysis information for it. (reg profile, instruction info, etc)

**Comment by slowhand99** [comment]
On it...

File: libr/asm/arch/jdh8/jdh8dis.c
Comment (slowhand99) [reply libr/asm/arch/jdh8/jdh8dis.c:35]: > #define to avoid having to pass buf and sizeof(buf)

How? Can you give an example?

Comment (trufae) [reply libr/asm/arch/jdh8/jdh8dis.c:35]: but dont u think is better to rewrite the plugin as an analysis one so it doesnt needs to duplicate the logic for the analysis? also, are there plans to implement the assembler fr this arch too?

**Review by trufae — CHANGES_REQUESTED** [review]
ill merge when the meson.build references it

**Review by trufae — CHANGES_REQUESTED** [review]
Can you add this disassembler for the meson builds and add a test? see libr/asm/meson.build and for the tests will be good to have a smple file for this architecture and post it in the radare2-testbins repo. is the bin format supported? because thats probably something to do too.

Im fine to merge this asm plugin, but it should be redone as an analysis plugin to avoid duplication of parsing logic as long as asm plugins do use anal ones when needed

**Comment by trufae** [comment]
Looks good to merge now! thank you! ill wait for the CI :)

File: libr/asm/meson.build
Comment (trufae) [inline libr/asm/meson.build:81]: uhm why remove it?

Comment (slowhand99) [reply libr/asm/meson.build]: Sorry mb... Pushed the previous commit

**Comment by condret** [comment]
Thanks!

Please use meaningful commit messages next time.

**Pull request merged**
